pkg/sinks/sink.go

```diff
@@ -246,6 +246,9 @@ func (u *SinkProcessor) Start(ctx context.Context) error {
 			forwardOpts = append(forwardOpts, sinkforward.WithCallbackUploader(cbPublisher))
 		}
 
+		// Derive the retryStrategy from the spec and add to the forwarder
+		forwardOpts = append(forwardOpts, sinkforward.WithRetryStrategy(u.VertexInstance.Vertex.Spec.Sink.RetryStrategy))
```

Comment:
Minor: we pass `u.VertexInstance` to `NewDataForward()`, which means everything in the spec is available there.